liquidator/index.js

```diff
@@ -128,22 +139,43 @@ async function run(logger, empAddress, pollingDelay, priceFeedConfig, liquidator
       });
     }
 
+    // Create a execution loop that will run indefinitely (or yield early if in serverless mode)
     while (true) {
-      // Get the current synthetic balance. Used to define the max tokens to liquidate.
-      const currentSyntheticBalance = await syntheticToken.balanceOf(accounts[0]);
-      // Update the liquidators state. This will update the clients, price feeds and gas estimator.
-      await liquidator.update();
-      // Check for liquidatable positions and submit liquidations. Bounded by current synthetic balance and considers the
-      // override price if the user has specified one.
-      await liquidator.liquidatePositions(currentSyntheticBalance, liquidatorOverridePrice);
-      // Check for any finished liquidations that can be withdrawn.
-      await liquidator.withdrawRewards();
-
+      await retry(
+        async () => {
+          // Update the liquidators state. This will update the clients, price feeds and gas estimator.
+          await liquidator.update();
+          // Check for liquidatable positions and submit liquidations. Bounded by current synthetic balance and
+          // considers override price if the user has specified one.
+          const currentSyntheticBalance = await syntheticToken.balanceOf(accounts[0]);
+          await liquidator.liquidatePositions(currentSyntheticBalance, liquidatorOverridePrice);
+          // Check for any finished liquidations that can be withdrawn.
+          await liquidator.withdrawRewards();
+        },
+        {
+          retries: errorRetries,
+          onRetry: error => {
+            logger.debug({
+              at: "Liquidator#index",
+              message: "An error was thrown in the execution loop - retrying",
```

Comment:
Maybe add here which retry out of max retry attempts you're on?

Reply:
I wanted to do that but it's not available in the this function, unfortunately. you can access it in the re-try loop but not in the `onRetry` call.

Reply:
huh ok I think its actually helpful enough to figure out a way to add it, perhaps we can pass into the `onRetry` callback somehow. WDYT @UMAprotocol/eng

Reply:
It's a little messy, but I think you could do something like:
```js
while(true) {
  let numRetries = 0;
  await retry(async () => {
    numRetires++;
    ...
  },
  {
    ...
    message: `retry ${numRetries} failed with error ${error}`
  });
}
```